### ISSUE TITLE
[React Native] Error on submitting the form - event.preventDefault() is not a function

### ISSUE THREAD

**Issue by ghoshabhi** [final-form/react-final-form#166]
### Are you submitting a **bug report** or a **feature request**?

Bug Report - similar to #83

### What is the current behavior?

I have a form like so: 
```
export default class extends Component<any> {
  displayName = 'MyForm';
  render() {
    return (
      <Form
        onSubmit={this.props.onSubmit}
        render={({ handleSubmit, values }) => (
          <View>
            <RowView // styled-component
              style={{ ... }}
            >
              <Text style={{ fontSize: 16, fontWeight: 'bold' }}>Label:</Text>
              <Field
                component={() => (
                  <TextInput
                    placeholder={'$00.00'}
                    keyboardType="numeric"
                    style={styles.textInput}
                  />
                )}
              />
            </RowView>
            <TouchableOpacity onPress={() => handleSubmit(values)}>
              <Text>Submit</Text>
            </TouchableOpacity>
          </View>
        )}
      />
    );
  }
}
```

Which I then use in a container:

```
class MyContainer extends Component {
  handleOnSubmit = (values) => {
   alert(JSON.stringify(values));
  };

  render() {
    return (
      <KeyboardAvoidingView style={styles.container}>
        <MyForm onSubmit={this.handleOnSubmit} />
      </KeyboardAvoidingView>
    );
  }
}
```

Unfortunately this throws an error when I try to submit the form, with the same error as mentioned in #83.

### What is the expected behavior?
Should submit form without any errors regarding the `event` object.

### Sandbox Link
https://snack.expo.io/r1Dj7J6vM  || https://snack.expo.io/S1IY_JawG

### What's your environment?

```
"react": "16.0.0-beta.5",
"react-final-form": "^3.1.0",
"react-native": "0.49.3",
```

### Other information

I see that error is coming from this function over [here](https://github.com/final-form/react-final-form/blob/master/src/ReactFinalForm.js#L113):

```
 _this.handleSubmit = function (event) {
      event && event.preventDefault();
      // console.log('event: ', JSON.stringify(event));
      return _this.form.submit();
    };
```
<img width="350" alt="screen shot 2018-02-22 at 4 29 09 pm" src="https://user-images.githubusercontent.com/19292575/36571829-9b3b7b5a-17ed-11e8-92ca-6d7d958bf7fd.png">

------
I am not really sure if I am doing anything wrong or using any of the components from the library wrongly.

**Comment by ghoshabhi** [comment]
Even though I see this issue was fixed and shipped with v2.2.0 - I am not sure why I am getting this issue then. I am sorry, if I am missing something very obvious.

**Comment by erikras** [comment]
The line in question is: [`event && event.preventDefault()`](https://github.com/final-form/react-final-form/blob/master/src/ReactFinalForm.js#L114).

In order for you to get that error, `event` must be truthy, but `event.preventDefault` not be a function. Perhaps it needs to be `event && event.preventDefault && event.preventDefault()`?

**Comment by ghoshabhi** [comment]
Is this published ? I tried to remove both the packages (`final-form` and `react-final-form`), but still see the same error. 

On a side note, I can submit the form now - but I cannot get the values from the `Form`  component ? Unlike `redux-form` this package doesn't need any hooking up with the reducers right ?

**Comment by erikras** [comment]
Not yet.

**Comment by erikras** [comment]
Published fix in [`v3.1.1`](https://github.com/final-form/react-final-form/releases/tag/v3.1.1).
